packages/cli/src/utils/index.js

```diff
@@ -10,30 +10,33 @@ import prettyBytes from 'pretty-bytes'
 import env from 'std-env'
 import { successBox, warningBox } from './formatting'
 
-export const requireModule = process.env.NUXT_TS ? require : esm(module, {
+const esmOptions = {
   cache: false,
   cjs: {
     cache: true,
     vars: true,
     namedExports: true
   }
-})
+}
 
 export const eventsMapping = {
   add: { icon: '+', color: 'green', action: 'Created' },
   change: { icon: env.windows ? '»' : '↻', color: 'blue', action: 'Updated' },
   unlink: { icon: '-', color: 'red', action: 'Removed' }
 }
 
-const getRootDir = argv => path.resolve(argv._[0] || '.')
-const getNuxtConfigFile = argv => path.resolve(getRootDir(argv), argv['config-file'])
-
-export async function loadNuxtConfig(argv) {
-  const rootDir = getRootDir(argv)
-  const nuxtConfigFile = getNuxtConfigFile(argv)
+export async function loadNuxtConfig(argv, extraOptions = {}) {
+  const rootDir = path.resolve(argv._[0] || '.')
+  let nuxtConfigFile = path.resolve(rootDir, argv['config-file'])
+  let requireModule = esm(module, esmOptions)
 
   let options = {}
 
+  if (extraOptions.typescript && !existsSync(nuxtConfigFile)) {
+    nuxtConfigFile = nuxtConfigFile.replace(/.js$/, '.ts')
```

Comment:
The `.ts` extension will be automatically added to the resolve method when using typescript runtime. There is no need to manually specify it unless something is wrong

Reply:
@pi0 The issue is that we need the `.js` or `.ts` explictly defined for the watchers

If we give `nuxt.config` to watchers, it will not watch the config

Reply:
If we can watch files that does not exist, we could make Nuxt watch `nuxt.config.js` / `nuxt.config.ts` Always, even if other config provided by CLI option

Reply:
@kevinmarrec i see your point. For this, we can resolve the correct file name using `require.resolve` :)

Reply:
nice, forgot this `require.resolve` thing, gonna update things, thanks for the tip !